python/python/tests/test_vector_index.py

```diff
@@ -465,3 +466,46 @@ def test_knn_with_deletions(tmp_path):
     assert len(results) == 10
 
     assert expected == [r.as_py() for r in results]
+
+
+def test_index_cache_size(tmp_path):
+    rng = np.random.default_rng(seed=42)
+
+    def query_index(ds, ntimes):
+        ndim = ds.schema[0].type.list_size
+        for _ in range(ntimes):
+            ds.to_table(
+                nearest={
+                    "column": "vector",
+                    "q": rng.standard_normal(ndim),
+                },
+            )
+
+    tbl = create_table(nvec=1024, ndim=16)
+    dataset = lance.write_dataset(tbl, tmp_path / "test")
+
+    indexed_dataset = dataset.create_index(
+        "vector",
+        index_type="IVF_PQ",
+        num_partitions=128,
+        num_sub_vectors=2,
+        index_cache_size=10,
+    )
+
+    assert (
+        indexed_dataset.stats.index_stats("vector_idx")["index_cache_entry_count"] == 1
+    )
+    query_index(indexed_dataset, 1)
+    assert (
+        indexed_dataset.stats.index_stats("vector_idx")["index_cache_entry_count"] == 2
+    )
+    query_index(indexed_dataset, 128)
+    assert (
+        indexed_dataset.stats.index_stats("vector_idx")["index_cache_entry_count"] == 10
+    )
+
+    indexed_dataset = lance.LanceDataset(indexed_dataset.uri, index_cache_size=50)
+    query_index(indexed_dataset, 256)
+    assert (
+        indexed_dataset.stats.index_stats("vector_idx")["index_cache_entry_count"] == 50
+    )
```

Comment:
Wouldn't this be a better test if we set `index_cache_size=5`, then range `query_index(indexed_dataset, 128)` and asserted `index_cache_entry_count` was just 5? That would actually show the limit is working, right? Otherwise, it's not 100% clear that we wouldn't get 50 back if we set no limit anyways.

Reply:
Sure! It also reduces CI work so changed.